admin-dev/themes/new-theme/js/pages/order/view/order-product-renderer.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import OrderViewPageMap from '@pages/order/OrderViewPageMap';
+import OrderProductEdit from '@pages/order/view/order-product-edit';
+import Router from '@components/router';
+
+const $ = window.$;
+
+export default class OrderProductRenderer {
+  constructor() {
+    this.router = new Router();
+  }
+
+  addOrUpdateProductToList($productRow, newRow) {
+    if ($productRow.length > 0) {
+      $productRow.html($(newRow).html());
+    } else {
+      $(OrderViewPageMap.productAddRow).before($(newRow).hide().fadeIn());
+    }
+  }
+
+  updateNumProducts(numProducts) {
+    $(OrderViewPageMap.productsCount).html(numProducts);
+  }
+
+  editProductFromList(orderDetailId, quantity, priceTaxIncl, priceTaxExcl, taxRate, location, availableQuantity, orderInvoiceId) {
+    const $orderEdit = new OrderProductEdit(orderDetailId);
+    $orderEdit.displayProduct({
+      price_tax_excl: priceTaxExcl,
+      price_tax_incl: priceTaxIncl,
+      tax_rate: taxRate,
+      quantity,
+      location,
+      availableQuantity,
+      orderInvoiceId,
+    });
+    $(OrderViewPageMap.productAddActionBtn).addClass('d-none');
+    $(OrderViewPageMap.productAddRow).addClass('d-none');
+  }
+
+  moveProductsPanelToModificationPosition(scrollTarget = 'body') {
+    this.moveProductPanelToTop();
+
+    $(OrderViewPageMap.productActionBtn).addClass('d-none');
+    $(`${OrderViewPageMap.productAddActionBtn}, ${OrderViewPageMap.productAddRow}`).removeClass('d-none');
+    const scrollValue = $(scrollTarget).offset().top - $('.header-toolbar').height() - 100;
```

Comment:
Why `-100` ? What does it refer to?

Reply:
Also maybe this scrolling part should be integrated in the `moveProductPanelToTop` function?

Reply:
The -100 is just there to add some space and not have the targeted element stuck exactly at the top of the screen. Yep, I'll add this at moveProductPanelToTop too 👍